lib/shower/index.js

```diff
@@ -1,3 +1,100 @@
-export { default as Container } from './container';
-export { default as defaultOptions } from './default-options';
-export { default as Shower } from './shower';
+import { EventTarget } from '../utils';
+import defaultOptions from './default-options';
+import Slide from './slide';
+import Player from './player';
+import Plugins from './plugins';
+
+const ensureId = (slideElement, index) => {
+    if (!slideElement.id) {
+        slideElement.id = index + 1;
+    }
+
+    return slideElement;
+};
+
+/**
+ * Core module of the Shower.
+ *
+ * @param {object} [options] Shower options.
+ */
+class Shower extends EventTarget {
+    constructor(options) {
+        super();
+        this.options = Object.freeze(Object.assign({}, defaultOptions, options));
+
+        const containerSelector = this.options.container_selector;
+        this.container = document.querySelector(containerSelector);
+        if (!this.container) {
+            throw new Error(`Shower container with selector '${containerSelector}' not found.`);
+        }
+
+        this._initPlayer();
+        this._initMode();
+        this.plugins = new Plugins(this);
+    }
+
+    _initPlayer() {
+        const slides = [ ...this.container.querySelectorAll(this.options.slides_selector) ]
+            .filter(slideElement => !slideElement.hidden)
+            .map(ensureId)
+            .map(slideElement => new Slide(slideElement, this.options.slide));
+
+        this.player = new Player(slides);
+    }
+
+    _initMode() {
+        if (this.container.classList.contains(this.options.mode_full_classname)) {
+            this.enterFullMode();
+        } else {
+            this.exitFullMode();
+        }
+    }
+
+    get isFullMode() {
+        return this._mode === 'full';
+    }
+
+    get isListMode() {
+        return this._mode === 'list';
+    }
```

Comment:
Shouldn't these properties be on the player? Or, as mentioned in https://github.com/shower/core/pull/134#issuecomment-366226493, should we have a SlideDeck model on which these properties can be read and set, with Player reacting to them?

Reply:
Also, given that the state is just boolean, why not simply `isFullScreen`?